### PR TITLE
dismiss related gregor items when tracking succeeds or is cancelled

### PR DESCRIPTION
**Pull request by oconnor663**
@maxtaco 

Preliminary PR, still need to implement the parts where we issue dismissals using this data.

### CODE REVIEW

File: go/libkb/identify_outcome.go [hunk 1]
```diff
@@ -10,20 +10,22 @@ import (
...
+	KeyDiffs              []TrackDiff
+	Revoked               []TrackDiffRevoked
+	RevokedDetails        []keybase1.RevokedProof
+	ProofChecks           []*LinkCheckResult
+	Warnings              []Warning
+	TrackUsed             *TrackLookup
+	TrackEqual            bool // Whether the track statement was equal to what we saw
+	TrackOptions          keybase1.TrackOptions
+	Reason                keybase1.IdentifyReason
+	ResponsibleGregorItem *gregor.Item
```
Comment (maxtaco) [inline go/libkb/identify_outcome.go]: since this is an interface, no need for `*`

File: go/engine/identify2_with_uid.go [hunk 2]
```diff
@@ -384,7 +387,9 @@ func (e *Identify2WithUID) getTrackChainLink(tmp bool) (*libkb.TrackChainLink, e
 }
 
 func (e *Identify2WithUID) createIdentifyState() (err error) {
-	e.state = libkb.NewIdentifyState(nil, e.them)
+	outcome := libkb.NewIdentifyOutcomeWithUsername(e.them.GetName())
+	outcome.ResponsibleGregorItem = e.responsibleGregorItem
+	e.state = libkb.NewIdentifyState(outcome, e.them)
```
Comment (maxtaco) [inline go/engine/identify2_with_uid.go]: maybe just pass the `ResponsibleGregorItem` into the interface?

**Comment by maxtaco** [comment]
Going in the right direction!

**Comment by oconnor663** [comment]
We're now exposing the gregorHandler through G using a new GregorDismisser interface defined in libkb. Using that, we implement the gregor dismissals that we've already made handlers for (https://github.com/keybase/client/pull/2833). Currently this will break tests because we haven't yet mocked our the GregorDismisser outside of the service.

File: go/service/gregor.go [hunk 3]
```diff
@@ -373,3 +375,18 @@ func (g *gregorHandler) connectNoTLS(uri *fmpURI) error {
 	g.conn = rpc.NewConnectionWithTransport(g, t, keybase1.ErrorUnwrapper{}, true, keybase1.WrapError, g.G().Log, nil)
 	return nil
 }
+
+func (g *gregorHandler) DismissItem(id gregor.MsgID) error {
+	idStruct := gregor1.MsgID(id.Bytes())
+	dismissal := gregor1.Message{
+		Ibm_: &gregor1.InBandMessage{
```
Comment (maxtaco) [inline go/service/gregor.go]: I think you need to specify metadata here too.

**Comment by maxtaco** [comment]
👍 Let's get this in, but I think a few things:
* we need tests (as discussed)
* we need metadata in the outgoing gregor message

**Comment by oconnor663** [comment]
Landed. Tests to come.

**Pull request closed without merging**